src/ui/public/vis/editors/default/keyboard_move.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * The keyboardMove directive can be attached to elements, that can receive keydown events.
+ * It will call the passed callback function and pass the direction in which an
+ * arrow key was pressed to the callback (as the argument with the name `direction`).
+ * The passed value will be one of `Direction.up` or `Direction.down`, which can be
+ * imported to compare against those values. The directive will also make sure, that
+ * the pressed button will get the focus back (e.g. if it was lost due to a ng-repeat
+ * reordering).
+ *
+ * Usage example:
+ *
+ * <button keyboard-move="onMoved(direction)">...</button>
+ *
+ * import { Direction } from './keyboard_move';
+ * function onMoved(dir) {
+ *   if (dir === Direction.up) {
+ *     // moved up
+ *   } else if (dir === Direction.down) {
+ *     // moved down
+ *   }
+ * }
+ */
+import { uiModules } from 'ui/modules';
+
+export const Direction = {
+  up: 'up',
+  down: 'down'
+};
+
+const directionMapping = {
```

Comment:
Makes totally sense. I just found, that we currently have mixed ways of exporting keycodes in the ui_framework services. That's why I created another PR (#13663). If you feel like that refactoring is a good thing, you could review that PR please, and I would rebase this PR as soon as I merged the other one to use the exported keycodes.